File: frontend/src/api/workspaceOverview.ts
import { apiGet } from '@/api/client'

export type OverviewTone = 'critical' | 'warning' | 'info'

export interface WorkspaceTimetable {
  id: number | null
  name: string
  status: string
  updated_at: string | null
  required_periods: number
  placed_periods: number
  remaining_periods: number
  completion_rate: number | null
}

export interface WorkspacePreflight {
  available: boolean
  error_count: number
  warning_count: number
  unavailable_message: string
}

export interface WorkspaceMetrics {
  active_teacher_count: number
  class_count: number
  weekly_affected_periods: number
  week_start: string
  week_end: string
}

export interface WorkspaceActionItem {
  code: string
  title: string
  description: string
  tone: OverviewTone
  target: string
  count: number | null
}

export interface WorkspaceOverview {
  semester_id: number
  semester_label: string
  generated_at: string
  metrics: WorkspaceMetrics
  timetable: WorkspaceTimetable
  preflight: WorkspacePreflight
  today_pending_periods: number
  unacknowledged_notifications: number
  focus_items: WorkspaceActionItem[]
  recommendations: WorkspaceActionItem[]
}

export const getWorkspaceOverview = (semesterId: number): Promise<WorkspaceOverview> =>
  apiGet(`/workspace-overview?semester_id=${semesterId}`)

File: backend/app/schemas/workspace_overview.py
"""工作空间首页总览的只读响应模型。"""

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field

OverviewTone = Literal["critical", "warning", "info"]


class WorkspaceTimetableOut(BaseModel):
    id: int | None = None
    name: str = ""
    status: str = ""
    updated_at: datetime | None = None
    required_periods: int = 0
    placed_periods: int = 0
    remaining_periods: int = 0
    completion_rate: int | None = None


class WorkspacePreflightOut(BaseModel):
    available: bool = True
    error_count: int = 0
    warning_count: int = 0
    unavailable_message: str = ""


class WorkspaceMetricsOut(BaseModel):
    active_teacher_count: int
    class_count: int
    weekly_affected_periods: int
    week_start: date
    week_end: date


class WorkspaceActionItemOut(BaseModel):
    code: str
    title: str
    description: str
    tone: OverviewTone
    target: str
    count: int | None = None


class WorkspaceOverviewOut(BaseModel):
    semester_id: int
    semester_label: str
    generated_at: datetime
    metrics: WorkspaceMetricsOut
    timetable: WorkspaceTimetableOut
    preflight: WorkspacePreflightOut
    today_pending_periods: int
    unacknowledged_notifications: int
    focus_items: list[WorkspaceActionItemOut] = Field(default_factory=list)
    recommendations: list[WorkspaceActionItemOut] = Field(default_factory=list)
